File: .template.zig
const std = @import("std");
const stdin = std.fs.File.stdin();
const stdout = std.fs.File.stdout();

pub fn main() !void {
    var r = Stdin.init(1024);
    var w = Stdout.init(1024);

    const n = try r.next(usize);
    const m = try r.last(usize);

    try w.print("{} {}\n", .{n, m});
}

// ==========================================
// HELPERS
// ==========================================

const Stdin = struct {
    reader: *std.Io.Reader,

    pub inline fn init(comptime N: usize) Stdin {
        var buf: [N]u8 = undefined;
        var reader = stdin.readerStreaming(&buf);
        return Stdin {
            .reader = &reader.interface,
        };
    }

    pub fn last(self: *Stdin, comptime T: type) !T {
        const token = try self.reader.takeDelimiterExclusive('\n');
        return try Stdin.parse(T, token);
    }

    pub fn next(self: *Stdin, comptime T: type) !T {
        const token = try self.reader.takeDelimiterExclusive(' ');
        return try Stdin.parse(T, token);
    }

    pub fn nextLine(self: *Stdin) ![]u8 {
        return try self.reader.takeDelimiterExclusive('\n');
    }

    pub fn nextList(self: *Stdin, comptime T: type, list: []T) !usize {
        const line = try self.nextLine();
        var it = std.mem.splitScalar(u8, line, ' ');

        var i: usize = 0;
        while (it.next()) |token| : (i += 1) {
            list[i] = try Stdin.parse(T, token);
        }

        return i;
    }

    fn parse(comptime T: type, token: []const u8) !T {
        if (T == i8 or T == i16 or T == i32 or T == i64 or T == i128 or T == isize) {
            return try std.fmt.parseInt(T, token, 10);
        } else if (T == u16 or T == u32 or T == u64 or T == u128 or T == usize) {
            return try std.fmt.parseInt(T, token, 10);
        } else if (T == f16 or T == f32 or T == f64 or T == f80 or T == f128) {
            return try std.fmt.parseFloat(T, token);
        } else if (T == bool) {
            return std.mem.eql(u8, token, "true");
        } else if (T == u8) {
            return token[0];
        } else {
            return token;
        }
    }
};

const Stdout = struct {
    writer: *std.Io.Writer,

    pub inline fn init(comptime N: usize) @This() {
        var buf: [N]u8 = undefined;
        var writer = stdout.writerStreaming(&buf);
        return .{
            .writer = &writer.interface,
        };
    }

    pub fn print(self: *@This(), comptime fmt: []const u8, args: anytype) !void {
        try self.writer.print(fmt, args);
        try self.writer.flush();
    }

    pub fn printList(self: *@This(), comptime T: type, comptime fmt: []const u8, args: []const T) !void {
        try self.writer.print(fmt, .{args[0]});
        for (1..args.len) |i| {
            try self.writer.writeByte(' ');
            try self.writer.print(fmt, .{args[i]});
        }
        try self.writer.print("\n", .{});
        try self.writer.flush();
    }
};

File: .template.rs
use std::fmt::Debug;
use std::io::{self, BufRead};
use std::str::FromStr;

fn main() {
    let n = last_int::<usize>();

    println!("{}", n);
}

//==========================================================
// Helpers
//==========================================================

#[allow(dead_code)]
fn last_float<T: Real>() -> T
where <T as FromStr>::Err: Debug
{
    let mut buffer = String::new();
    io::stdin().read_line(&mut buffer).expect("%f\n");

    buffer.trim().parse::<T>().expect("parseable as float")
}

#[allow(dead_code)]
fn last_int<T: Num>() -> T
where <T as FromStr>::Err: Debug
{
    let mut buffer = String::new();
    io::stdin().read_line(&mut buffer).expect("%d\n");

    buffer.trim().parse::<T>().expect("parseable as int")
}

#[allow(dead_code)]
fn next() -> String {
    let mut buffer: Vec<u8> = Vec::new();

    io::stdin().lock().read_until(b' ', &mut buffer).expect("contains space");

    String::from_utf8(buffer).expect("parseable as utf8")
}

#[allow(dead_code)]
fn next_float<T: Num>() -> T
where <T as FromStr>::Err: Debug
{
    next().trim().parse::<T>().expect("parseable as float")
}

#[allow(dead_code)]
fn next_int<T: Num>() -> T
where <T as FromStr>::Err: Debug
{
    next().trim().parse::<T>().expect("parseable as int")
}

#[allow(dead_code)]
fn next_line() -> String {
    let mut buffer = String::new();
    io::stdin().read_line(&mut buffer).expect("%s\n");

    buffer
}

#[allow(dead_code)]
fn next_list<T: Real>(n: usize) -> Vec<T>
where <T as FromStr>::Err: Debug {
    let mut vec: Vec<T> = Vec::new();

    while vec.len() < n {
        let mut buffer = String::new();
        io::stdin().read_line(&mut buffer).expect("%s\n");


        buffer
            .trim()
            .split(' ')
            .map(|s| s.parse::<T>().expect("parseable as any number type"))
            .for_each(|num| vec.push(num))
    }

    vec
}

#[allow(dead_code)]
fn next_tuple<T: Real>() -> (T, T)
where <T as FromStr>::Err: Debug {
    let mut buffer = String::new();
    io::stdin().read_line(&mut buffer).expect("%s\n");

    let mut tuple = buffer
        .trim()
        .split(' ')
        .map(|s| s.parse::<T>().expect("parseable as any number type"));

    (
        tuple.next().expect("(?,)"),
        tuple.next().expect("(,?)"),
    )
}

//==========================================================
// Type System
//==========================================================

#[allow(dead_code)]
trait Real: Sized + FromStr + Debug {}
#[allow(dead_code)]
trait Num: Real {}
#[allow(dead_code)]
trait Int: Num {}
#[allow(dead_code)]
trait Nat: Num {}

macro_rules! is_int {
    ($alias:ty) => {
        impl Real for $alias {}
        impl Num for $alias {}
        impl Int for $alias {}
    }
}
macro_rules! is_nat {
    ($alias:ty) => {
        impl Real for $alias {}
        impl Num for $alias {}
        impl Nat for $alias {}
    }
}

macro_rules! is_real {
    ($alias:ty) => {
        impl Real for $alias {}
    }
}

is_int!(i16);
is_int!(i32);
is_int!(i64);
is_int!(i128);
is_int!(isize);

is_nat!(u16);
is_nat!(u32);
is_nat!(u64);
is_nat!(u128);
is_nat!(usize);

is_real!(f32);
is_real!(f64);

//==========================================================
